### PR TITLE
Issues/5853

### PR DESCRIPTION
**Pull request by NorseGaud**
Handling noexec on /tmp using a local tmp folder
Added logic to prevent overuse of resources in make -j

### CODE REVIEW

File: eosio_build.sh [hunk 1]
```diff
@@ -268,7 +276,7 @@
       printf "\\n\\t>>>>>>>>>>>>>>>>>>>> EOSIO has been successfully configured but not yet built.\\n\\n"
       exit 0
    fi
-
+   if [ $CPU_CORE -gt 4 ]; then CPU_CORE=$((CPU_CORE / 2)); fi
```
[the remark this answers is not in the document]
Comment (NorseGaud) [reply eosio_build.sh]: I've tried two different fresh Centos 7 machines with 16GB of memory and while building, it throws the error shown in https://github.com/EOSIO/eos/issues/4573

Reducing the total jobs well below the required amount slows the build process, but makes the build script more compatible with the different setups people have. 

I'm sure there is a better way of accomplishing this so I'm open to suggestions.

[the remark this answers is not in the document]
Comment (NorseGaud) [reply eosio_build.sh]: Here is my setup if it helps:

	CPU cores: 24
	Physical Memory: 15843 Mgb

[the remark this answers is not in the document]
Comment (spoonincode) [reply eosio_build.sh]: The problem with bringing in #4573 to the argument is that I think there are a few files that take more than 4GB to compile with g++. Even at `make -j1`. (I think I needed to turn on swap on a c5.large instance to get it to build).

I would prefer to continue to limit the parallelism within the centos, aws, and fedora scripts. Ubuntu and macos use clang to build and it easily gets through `make -j16` with even just 16GB of RAM. It looks like the scripts already limit `-j` to half the number of gigabytes available. Maybe you can try changing the script to further limit it to a quarter?

**Comment by NorseGaud** [comment]
Ok, moving this back to just issue/5853. @spoonincode, I got a great idea for the resource usage problem that I'll open a new PR for. Thanks for the help!

**Comment by heifner** [comment]
Closing this issue as it has been replaced by #5864.

**Pull request closed without merging**